yarn-project/noir-contracts/src/contracts/schnorr_auth_witness_account_contract/src/main.nr

```diff
@@ -0,0 +1,36 @@
+mod util;
+mod auth_oracle;
+
+contract SchnorrAuthWitnessAccount {
+    use dep::std::hash::pedersen_with_separator;
+    use dep::aztec::entrypoint::EntrypointPayload;
+    use dep::aztec::constants_gen::GENERATOR_INDEX__SIGNATURE_PAYLOAD;
+    use crate::util::recover_address;
+    use crate::auth_oracle::get_auth_witness;
+
+    #[aztec(private)]
+    fn constructor() {}
+
+    #[aztec(private)]
+    fn entrypoint(
+        payload: pub EntrypointPayload,
+    ) {
+        let message_hash: Field = pedersen_with_separator(
+          payload.serialize(), 
+          GENERATOR_INDEX__SIGNATURE_PAYLOAD
+        )[0];
+        let witness = get_auth_witness(message_hash);
+        assert(recover_address(message_hash, witness) == context.this_address());
```

Comment:
I'm guessing you cannot call `is_valid` directly to be inlined because it's already an external function, so it is initialising its own PrivateContext, right?

Reply:
Will create an `inner` function that they are both calling, a little clunky, but should work fine for adding a public is valid later on as well.